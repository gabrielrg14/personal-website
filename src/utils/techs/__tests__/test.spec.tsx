import {
    knowledgeTechs,
    knownTechs,
    potentialTechs,
    studyingTechs,
    techListing
} from "utils"

describe("techs", () => {
    describe("knowledgeTechs", () => {
        it("should match snapshot", () => {
            expect(knowledgeTechs).toMatchSnapshot()
        })

        it("should have a length greater than 0", () => {
            expect(knowledgeTechs.length).toBeGreaterThan(0)
        })
    })

    describe("knownTechs", () => {
        it("should match snapshot", () => {
            expect(knownTechs).toMatchSnapshot()
        })

        it("should have a length greater than 0", () => {
            expect(knownTechs.length).toBeGreaterThan(0)
        })
    })

    describe("potentialTechs", () => {
        it("should match snapshot", () => {
            expect(potentialTechs).toMatchSnapshot()
        })

        it("should have a length greater than 0", () => {
            expect(potentialTechs.length).toBeGreaterThan(0)
        })
    })

    describe("studyingTechs", () => {
        it("should match snapshot", () => {
            expect(studyingTechs).toMatchSnapshot()
        })

        it("should have a length greater than 0", () => {
            expect(studyingTechs.length).toBeGreaterThan(0)
        })
    })

    describe("techListing", () => {
        it("should have a length greater than 0 and contain the icon, label and identifier properties", () => {
            expect(techListing.length).toBeGreaterThan(0)
            expect(techListing[0]).toHaveProperty("icon")
            expect(techListing[0]).toHaveProperty("label")
            expect(techListing[0]).toHaveProperty("identifier")
        })
    })
})
