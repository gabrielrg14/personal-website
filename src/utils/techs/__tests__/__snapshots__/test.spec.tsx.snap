// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`techs knowledgeTechs should match snapshot 1`] = `
[
  "react-native",
  "expo",
  "vuejs",
  "redux",
  "graphql",
  "nodejs",
  "tailwindcss",
]
`;

exports[`techs knownTechs should match snapshot 1`] = `
[
  "html",
  "css",
  "javascript",
  "typescript",
  "react",
  "nextjs",
  "styled-components",
  "bootstrap",
  "jest",
  "testing-library",
  "git",
  "github",
]
`;

exports[`techs potentialTechs should match snapshot 1`] = `
[
  "cypress",
  "nestjs",
  "aws",
  "azure",
]
`;

exports[`techs studyingTechs should match snapshot 1`] = `
[
  "docker",
  "kubernetes",
]
`;
